src/main/java/com/ably/kafka/connect/client/DefaultAblyClient.java

```diff
@@ -59,6 +67,21 @@ public void publishFrom(SinkRecord record) throws ConnectException {
             //this exception should cause the calling task to abort
             throw new ConnectException("Cannot publish to Ably when connection failed");
         }
+        //check both channel and message config for skipping the record
+        final String messageConfig = connectorConfig.getString(MESSAGE_CONFIG);
+        final String channelConfig = connectorConfig.getString(CHANNEL_CONFIG);
+        final boolean skipOnKeyAbsence = connectorConfig.getBoolean(SKIP_ON_KEY_ABSENCE);
+        if (skipOnKeyAbsence){
+            final ConfigValueEvaluator.Result messageResult = configValueEvaluator.evaluate(record, messageConfig, true);
+            final ConfigValueEvaluator.Result channelResult = configValueEvaluator.evaluate(record, channelConfig, true);
+
+            if (messageResult.shouldSkip() || channelResult.shouldSkip()) {
+                logger.warn("Skipping record as record key is not available in a record where the config for either" +
```

Comment:
The split and subsequent need for string concatenation here seems a bit arbitrary and unnecessary. I would just put this whole statement on one line. 🤷

Reply:
This split is based on line length set on IDE. You can see this elsewhere in this codebase, for example `ChannelSinkConnectorConfig` will have similar splits.

Reply:
I'm increasingly of the opinion that any line length limit is arbitrary and will always be subjective. Is your IDE instructed by any configuration in this codebase? Or is this your own local configuration and/or IDE default? Would exceeding this limit have failed in CI checks?

My preference would be for you to circumvent or otherwise disable this check so that the code can be laid out logically.

Reply:
> Or is this your own local configuration and/or IDE default?


.editorconfig is set on this project. But there is not any enforced line length in it, So editor shows its default line length. Adding max line length setting will just change the line the horizontal spacing from start line and will not effect re-format. Please see:
<img width="1201" alt="Screenshot 2022-10-31 at 17 47 23" src="https://user-images.githubusercontent.com/421091/199075110-a4d5fe71-ed92-439c-a8b2-196068a7a644.png">

So the answer is that this is not automatic and there is no CI checks on this repo which will fail if there is an issue with formatting, but also not completely arbitrary too. As I commented on the top, a similar line length has been followed throughout this codebase. Can we keep this like this now until we set up a lint checker on our CI?

Reply:
Sure and thanks for explaining. 😁 ... I'm happy to leave as is for now and address later on when we come to setting up check style or other lint tool.